### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by CSMastermind**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    // Object members
+    __id__: String;
+    amorphicLeaveEmpty: boolean;
+    amorphicToJSON(callback? : Function) : string;
+    amorphicGetPropertyDefinition(propertyName : string);
+    amorphicGetPropertyValues(propertyName : string);
+    amorphicGetPropertyDescriptions(propertyName : string);
+
+    // Deprecated legacy naming
+    static createProperty(prop : String, defineProperty: Object)
```
Comment (hackerrdave) [inline index.d.ts]: can we standardize to not have space prior to `:`?

File: test/typescript/model/Bear.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import {Supertype, supertypeClass, property} from '../../../index';
+import {Animal} from './Animal';
+
+@supertypeClass
+export class Bear extends Animal
+{
```
Comment (hackerrdave) [inline test/typescript/model/Bear.ts]: can we standardize to have all class definition `{` braces on same line?
```javascript
export class Bear extends Animal {
```

File: index.d.ts (hunk 1 above)
Comment (CSMastermind) [inline index.d.ts]: Yep!  Fixing now.

File: test/typescript/model/Bear.ts (hunk 2 above)
Comment (CSMastermind) [inline test/typescript/model/Bear.ts]: Seems like a reasonable extension of our current rules (fixed).

**Pull request merged**